src/pages/MyPatches.tsx

```diff
@@ -1,216 +1,14 @@
-            { fetchMoreResult }: { fetchMoreResult: UserPatchesQuery }
-          ) => {
-            if (!fetchMoreResult) {
-              return prev;
-            }
-            return fetchMoreResult;
-          },
-        });
-      } catch (e) {
-        // empty block
-      }
-    };
-    fetch();
-  }, [search, fetchMore]);
-
-  const onCheckboxChange = (): void => {
-    replace(
-      `${pathname}?${queryString.stringify(
-        {
-          ...queryString.parse(search, { arrayFormat }),
-          [MyPatchesQueryParams.CommitQueue]: !getQueryVariables(search)
-            .includeCommitQueue,
-        },
-        { arrayFormat }
-      )}`
-    );
-  };
-
-  const renderTable = () => {
-    if (error) {
-      return (
-        <PageWrapper>
-          <Banners
-            banners={bannersState}
-            removeBanner={dispatchBanner.removeBanner}
-          />
-        </PageWrapper>
-      );
-    }
-    if (loading) {
-      return <StyledSkeleton active title={false} paragraph={{ rows: 4 }} />;
-    }
-    if (get(data, "userPatches.patches", []).length !== 0) {
-      return data.userPatches.patches.map((p) => (
-        <PatchCard key={p.id} {...p} />
-      ));
-    }
-    return <NoResults data-cy="no-patches-found">No patches found</NoResults>;
-  };
-
-  const { limit, page, includeCommitQueue } = getQueryVariables(search);
-  return (
-    <PageWrapper>
-      <Banners
-        banners={bannersState}
-        removeBanner={dispatchBanner.removeBanner}
-      />
-      <PageTitle>My Patches</PageTitle>
-      <FiltersWrapperSpaceBetween>
-        <FlexRow>
-          <StyledInput
-            placeholder="Search Patch Descriptions"
-            onChange={patchNameFilterValueOnChange}
-            suffix={<Icon glyph="MagnifyingGlass" />}
-            value={patchNameFilterValue}
-            data-cy="patch-description-input"
-            width="25%"
-          />
-          <StatusSelector />
-        </FlexRow>
-        <StyledCheckbox
-          data-cy="commit-queue-checkbox"
-          onChange={onCheckboxChange}
-          label="Show Commit Queue"
-          checked={includeCommitQueue}
-        />
-      </FiltersWrapperSpaceBetween>
-      <PaginationRow>
-        <Pagination
-          pageSize={limit}
-          value={page}
-          totalResults={get(data, "userPatches.filteredPatchCount", 0)}
-          dataTestId="my-patches-pagination"
-        />
-        <PageSizeSelector
-          dataTestId="my-patches-page-size-selector"
-          value={limit}
-        />
-      </PaginationRow>
-      <>{renderTable()}</>
-    </PageWrapper>
-  );
+import { GetUserQuery } from "gql/generated/types";
+import { GET_USER } from "gql/queries";
+import { useHistory } from "react-router-dom";
+import { getUserPatchesRoute } from "constants/routes";
+
+export const MyPatches: React.FC = () => {
+  const router = useHistory();
+  const { data } = useQuery<GetUserQuery>(GET_USER);
```

Comment:
this is making an additional request for the same query as the top level query in the routes component. this should either be deleted altogether or subscribe to the cache value of userId by making a local query. I think doing both is actually not a bad idea.

Reply:
the fetchPolicy defaults to cache-first. so if the data has been fetched and exists in the cache, it will be utilized. otherwise a network request is made.